webapi/src/test/scala/org/knora/webapi/responders/v2/ResourcesResponderV2Spec.scala

```diff
@@ -2384,7 +2384,110 @@ class ResourcesResponderV2Spec extends CoreSpec() with ImplicitSender {
           assert(entityUsedResponse.results.bindings.isEmpty, s"Link value was not erased")
       }
     }
+  }
+  "When given a custom IRI" should {
+
+    "create a resource with no values but a custom IRI" in {
+      // Create the resource.
+
+      val resourceIri: IRI = "http://rdfh.ch/0001/55UrkgTKR2SEQgnsLWI9kk"
+
+      val inputResource = CreateResourceV2(
+        resourceIri = Some(resourceIri.toSmartIri),
+        resourceClassIri = "http://0.0.0.0:3333/ontology/0001/anything/v2#Thing".toSmartIri,
+        label = "thing with a custom IRI",
+        values = Map.empty,
+        projectADM = SharedTestDataADM.anythingProject
+      )
+
+      responderManager ! CreateResourceRequestV2(
+        createResource = inputResource,
+        featureFactoryConfig = defaultFeatureFactoryConfig,
+        requestingUser = anythingUserProfile,
+        apiRequestID = UUID.randomUUID
+      )
+
+      // Check that the response contains the correct metadata.
+
+      expectMsgPF(timeout) {
+        case response: ReadResourcesSequenceV2 =>
+          val outputResource: ReadResourceV2 = response.toResource(resourceIri).toOntologySchema(ApiV2Complex)
+
+          checkCreateResource(
+            inputResourceIri = resourceIri,
+            inputResource = inputResource,
+            outputResource = outputResource,
+            defaultResourcePermissions = defaultAnythingResourcePermissions,
+            defaultValuePermissions = defaultAnythingValuePermissions,
+            requestingUser = anythingUserProfile
+          )
+      }
+
+      // Get the resource from the triplestore and check it again.
```

Comment:
Why do we need to check a newly created resource twice? Isn't this already done on the previous check?

Reply:
In the first check, the resource is converted to the complex schema to check that its metadata is correct.
In the second check, we get the created resource by its IRI and check its values.

Reply:
ok, thanks